### DISCUSSION TITLE
Do Not Disturb cannot see subscribed .ics calendars on Android 12?

### DISCUSSION THREAD

**Discussion by emacsomancer** [bitfireAT/icsx5#40] [Q&A]
On an Android 10 device I was able to subscribe to an .ics calendar via ICSx5 and have Do Not Disturb turn on during calendar events from that calendar, but on my Android 12/GrapheneOS device Do Not Disturb doesn't seem to be able to 'see' that calendar (though it can see locally created calendars).

**Comment by javiertury** [comment]
Same here. For me this bug has been present in Android 11 and now in Android 12.

**Comment by ArnyminerZ** [comment]
Unfortunately, this looks like a Google Calendar-only feature (see [docs](https://support.google.com/android/answer/9069335?hl=en#zippy=%2Cduring-events-meetings-on-your-google-calendar)). So there's pretty much nothing we can do about it 😞.

Let us know if you know any reference on this being done on 3rds-calendars.

**Reply by javiertury** [reply]
I also use DAVx5 for caldav and those calendars appear in the "do not disturb" selector.

**Reply by ArnyminerZ** [reply]
Yup, just noticed that. I will take a further look into it.

**Comment by javier-garcia-tilburg** [comment]
I found an stack overflow question about the same problem, [Automatic Do Not Disturb rules for calendar events do not work with imported calendar](https://android.stackexchange.com/questions/155055/automatic-do-not-disturb-rules-for-calendar-events-do-not-work-with-imported-cal).

The logic about what events trigger "do not disturb" seems to be in the file [CalendarTracker.java](https://android.googlesource.com/platform/frameworks/base/+/master/services/core/java/com/android/server/notification/CalendarTracker.java), in the methods [checkEvent()](https://android.googlesource.com/platform/frameworks/base/+/master/services/core/java/com/android/server/notification/CalendarTracker.java#119) and [getCalendarsWithAccess()](https://android.googlesource.com/platform/frameworks/base/+/master/services/core/java/com/android/server/notification/CalendarTracker.java#92).

The logic about what calendars can be added to "do not disturb" seems to be in the file [ZenModeEventRuleSettings.java](https://cs.android.com/android/platform/superproject/+/master:packages/apps/Settings/src/com/android/settings/notification/zen/ZenModeEventRuleSettings.java), in the methods [getCalendars()](https://cs.android.com/android/platform/superproject/+/master:packages/apps/Settings/src/com/android/settings/notification/zen/ZenModeEventRuleSettings.java;l=187;drc=a21ac03acb5a25caf4165d9fae3b39f43b8d006b) and [addCalendars()](https://cs.android.com/android/platform/superproject/+/master:packages/apps/Settings/src/com/android/settings/notification/zen/ZenModeEventRuleSettings.java;l=207;drc=a21ac03acb5a25caf4165d9fae3b39f43b8d006b).

**Reply by ArnyminerZ** [reply]
Yeah, the issue is that for some reason this feature checks for write permissions on the calendars. Since ICSx⁵ calendars are read-only, DND considers them as "not valid". The only way of fixing this without Google changing anything would be giving write access to the calendars, which would break some things, since the objective of ICSx⁵ is on read-only calendars.

**Reply by englut** [reply]
@ArnyminerZ what if you gave us an option to set the calendars as RW and mark the option as experimental?

**Reply by ArnyminerZ** [reply]
It would be unproductive I think. I agree that we could add a toggle to tell the system that the calendar is RW, but changes will not be persisted, and it may get confusing to some users.

**Reply by englut** [reply]
I hear ya, it's weird. I do have a gcal calendar I subscribe to that I have write permissions to (its my work calendar).. But icsx5 doesn't allow me to write to it. This is the calendar I'm wanting to have DnD mode turned on during events for.
